### PR TITLE
Extended application matching

### PR DESCRIPTION
**Pull request by TheMarex**
I made it possible to match for specific attributes like class and instance. Now application rules as follows are possible:

```
{
    match = {
        class = {
            "Do",
        },
        "clock%-applet",
    },
    float = true,
    intrusive = true,
},
```

### CODE REVIEW

**Comment by bioe007** [comment]
This would close issue #2

I'm in favor but need to rewrite the readme and [wiki](http://awesome.naquadah.org/wiki/Shifty) 

I can probably manage that this week, unless someone else will :) 

Perhaps even a message in awesome mailing list.

**Comment by alfredopalhares** [comment]
Ok, gave a look at the code. Its nice. Will test it on my setup and documente on the readme and wiki.

File: init.lua [hunk 1]
```diff
@@ -475,80 +475,133 @@ function match(c, startup)
...
-                    end
-                    if a.float ~= nil then float = a.float end
-                    if a.slave ~=nil then slave = a.slave end
-                    if a.border_width ~= nil then
-                        c.border_width = a.border_width
+            local matched = false
+            -- match only class
+            if not matched and cls and a.match.class then
+                for k, w in ipairs(a.match.class) do
+                    matched = cls:find(w)
```
Comment (alfredopalhares) [inline init.lua:482]: I personally dont like this syntax sugar things. It s inconsistent  in this library tough. So i guess it OK.

Comment (bioe007) [inline init.lua:482]: You mean the colon for member functions? I would say there is a lot of other ugliness in the code...  

w0w.. didn't even know we could do this commenting in lines. O.O

Comment (alfredopalhares) [inline init.lua:482]: Yeah its pretty sweet to discuss code :D

**Pull request merged**